sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala

```diff
@@ -723,8 +723,12 @@ class DataSourceV2SQLSuite
     df.createOrReplaceTempView("source")
 
     sql(s"CREATE TABLE table_name USING parquet AS SELECT id, data FROM source")
-
-    checkAnswer(sql(s"TABLE default.table_name"), spark.table("source"))
+    val x = spark.table("source")
+    val xr = x.collect()
+    val y = sql(s"TABLE default.table_name")
+    val xy = y.collect()
```

Comment:
val xy will be empty but it shouldn't be.

Reply:
cc @cloud-fan

Reply:
I think it's a test problem. `spark.conf.unset(V2_SESSION_CATALOG_IMPLEMENTATION.key)` doesn't work anymore as the session catalog instance is in the map now. We should call `CatalogManager.reset` after `spark.conf.unset(V2_SESSION_CATALOG_IMPLEMENTATION.key)`